core/include/cubos/core/reflection/traits/fields.hpp

```diff
@@ -54,19 +54,30 @@ namespace cubos::core::reflection
         /// @param type Field type.
         /// @param name Field name.
         /// @param addressOf Getter for the address of the field on a given instance.
+        void addField(const Type& type, std::string name, AddressOf* addressOf);
+
+        /// @copydoc addField(const Type&, std::string, AddressOf*)
         /// @return Trait.
         FieldsTrait&& withField(const Type& type, std::string name, AddressOf* addressOf) &&;
```

Comment:
Why is this needed? The templated `withField` calls `addField`

Reply:
We use in the `ComponentTypeBuilder` to avoid having to do this:
```cpp
mFieldsTrait = std::move(mFieldsTrait).withField(...);
```
This would mean having to implement the assignment operator for `FieldsTrait`. This way we can just do:
```cpp
mFieldsTrait.addField(...);
```

Reply:
The review was about the 3 argument `withField`, as I tought it was dead code as both templated `addField` and `withField` used the 3 argument `addField`.
This isn't the case, glm reflection uses 3 argument `withField` so it makes sense to keep it.